lib/graphql/tracing/data_dog_tracing.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Tracing
+    class DataDogTracing < PlatformTracing
+      self.platform_keys = {
+        'lex' => 'lex.graphql',
+        'parse' => 'parse.graphql',
+        'validate' => 'validate.graphql',
+        'analyze_query' => 'analyze.graphql',
+        'analyze_multiplex' => 'analyze.graphql',
+        'execute_multiplex' => 'execute.graphql',
+        'execute_query' => 'execute.graphql',
+        'execute_query_lazy' => 'execute.graphql',
+      }
+
+      def platform_trace(platform_key, key, data)
+        pin = Datadog::Pin.get_from(self)
+        unless pin
+          pin = Datadog::Pin.new('graphql-ruby', app: 'graphql-ruby')
+          pin.onto(self)
+        end
+
+        pin.tracer.trace(platform_key, service: pin.service) do |span|
+          if key == 'execute_multiplex'
+            span.resource = data[:multiplex].queries.map(&:selected_operation_name).join(', ')
+          end
+
+          if key == 'execute_query'
+            span.set_tag(:selected_operation_name, data[:query].selected_operation_name)
+            span.set_tag(:selected_operation_type, data[:query].selected_operation.operation_type)
+            span.set_tag(:query_string, data[:query].query_string)
+            span.set_tag(:variables, data[:query].variables.to_h)
```

Comment:
While awfully useful, this could be a bit dangerous! Imagine a mutation like "log in", for example. Wouldn't this cause you to run the risk of storing the user's credentials in DataDog? Even worse, imagine a mutation that passes along a user's payment details, such as a creditcard!

Reply:
@esjee what do You think about current solution? :)